### PR TITLE
Unit tests for the function 'check_mozregression_version' in mozregression.main.py

### PR DESCRIPTION
**Pull request by dangerouspython**
It would be better to have some unit tests for the function 'check_mozregression_version' in mozregression.main.py. Thats why I wrote these functions

### CODE REVIEW

File: tests/unit/test_main.py [hunk 1]
```diff
@@ -246,5 +248,69 @@ def test_preference_args(self):
         prefs = main.preference(None, prefs_args)
         self.assertEquals(len(prefs), 0)
 
+
+class TestCheckMozregresionVersion(unittest.TestCase):
+    @patch('requests.get')
+    @patch('sys.stdout')
+    def test_pypi_is_accesible(self, stdout, get):
```
Comment (parkouss) [inline tests/unit/test_main.py]: Hmm, I don't understand the name of this function. Shouldn'it be something like "test_version_is_up_to_date" ?

File: tests/unit/test_main.py [hunk 2]
```diff
@@ -246,5 +248,69 @@ def test_preference_args(self):
...
+        options = None
+        logger = Mock('mozlog.structured.commandline.setup_logging')(
```
Comment (parkouss) [inline tests/unit/test_main.py]: I think that just looger = Mock() is sufficient here. Plus, you won't have to mock stdout.

Comment (dangerouspython) [inline tests/unit/test_main.py]: Ok.. Sure! Will re-push it again in a few seconds...

File: tests/unit/test_main.py [hunk 3]
```diff
@@ -246,5 +248,69 @@ def test_preference_args(self):
...
+    @patch('requests.get')
+    @patch('sys.stdout')
+    def test_RequestException_error(self, stdout, get):
+        options = None
+        logger = Mock('mozlog.structured.commandline.setup_logging')(
+                      "mozregression",
+                      options,
+                      {"mach": stdout})
+        get.side_effect = requests.RequestException
+        main.check_mozregression_version(logger)
```
Comment (parkouss) [inline tests/unit/test_main.py]: could you add a comment above this to indicate that the exception is handled inside main.check_mozregression_version ?

File: tests/unit/test_main.py [hunk 4]
```diff
@@ -246,5 +248,69 @@ def test_preference_args(self):
...
+        self.assertRaises(requests.RequestException, get)
+
+    @patch('requests.get')
+    @patch('sys.stdout')
+    def test_key_error(self, stdout, get):
```
Comment (parkouss) [inline tests/unit/test_main.py]: Well, this seems not really useful to me. the code executed is the same for the previous test. If you want to test every kind of handled exception, maybe write a loop in one test method. or jut get rid of this test method :)

File: tests/unit/test_main.py [hunk 5]
```diff
@@ -246,5 +248,69 @@ def test_preference_args(self):
...
+                      "mozregression",
+                      options,
+                      {"mach": stdout})
+        get.side_effect = KeyError
+        main.check_mozregression_version(logger)
+        self.assertRaises(KeyError, get)
+
+    @patch('requests.get')
+    @patch('sys.stdout')
+    def test_value_error(self, stdout, get):
```
Comment (parkouss) [inline tests/unit/test_main.py]: Same as above I think.

File: tests/unit/test_main.py [hunk 6]
```diff
@@ -246,5 +248,69 @@ def test_preference_args(self):
...
+                      "mozregression",
+                      options,
+                      {"mach": stdout})
+        get.side_effect = ValueError
+        main.check_mozregression_version(logger)
+        self.assertRaises(ValueError, get)
+
+    @patch('requests.get')
+    @patch('sys.stdout')
+    def test_warning_is_called_twice(self, stdout, get):
```
Comment (parkouss) [inline tests/unit/test_main.py]: I don't really like the name here. Maybe "test_warn_if_version_not_up_to_date" ?

**Comment by parkouss** [comment]
Hey @dangerouspython, thanks for this ! I noted some thing on the PR, could you address them please ?

**Pull request merged**